tracers/tracers.go

```diff
@@ -82,7 +87,14 @@ func (d *directory) RegisterJSEval(f jsCtorFn) {
 // New returns a new instance of a tracer, by iterating through the
 // registered lookups. Name is either name of an existing tracer
 // or an arbitrary JS code.
-func (d *directory) New(name string, cfg json.RawMessage, allowCustom bool) (Tracer, error) {
+func (d *directory) New(name string, cfg json.RawMessage, allowCustom bool) (tr Tracer, err error) {
+	defer func() {
```

Comment:
Not necessary

Reply:
This is to cover the case the body request has a specific tracer name (in the case `opt.Name` is not an empty string we allow custom tracers), but it doesn't exist. In that case this [line](https://github.com/vechain/thor/blob/8a65c7f6bcc182bc889b982a09aaf934866a3021/tracers/tracers.go#L108) will panic

Reply:
Sorry I am not following, could you give a case that the specific line will panic?

Reply:
calling the `debug/tracers` endpoint with a payload like in [this](https://github.com/vechain/thor/blob/8a65c7f6bcc182bc889b982a09aaf934866a3021/api/debug/debug_test.go#L165) test will panic, but actually calling the endpoint against a real node instance returns an error. Not sure why this happens, probably the `jsEval` method hides some context that is missing during a test run. I'll remove it, then

Reply:
This is a tricky one, panic is caused by not loading js tracer engine, you need to add `_"github.com/vechain/thor/v2/tracers/js"` to the import section

Reply:
Looks like you have removed the `nonExistTracer` test in https://github.com/vechain/thor/pull/686/commits/adcfa604b1666b1c0aa05e8e29efa9c7fd7777c5, do you have a plan adding it back or just leave it as is?

Reply:
I think I'll add it back, it doesn't deserve to be deleted after all 😆